core/modules/ccontrol/UserQueryProcessList.cc

```diff
@@ -0,0 +1,242 @@
+#include "sql/SqlErrorObject.h"
+#include "sql/SqlBulkInsert.h"
+#include "sql/statement.h"
+#include "util/IterableFormatter.h"
+
+using namespace lsst::qserv;
+
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.ccontrol.UserQueryProcessList");
+
+std::string g_nextResultTableId(std::string const& userQueryId) {
+    return "qserv_result_processlist_" + userQueryId;
+}
+
+}
+
+namespace lsst {
+namespace qserv {
+namespace ccontrol {
+
+// Constructor
+UserQueryProcessList::UserQueryProcessList(std::shared_ptr<query::SelectStmt> const& statement,
+        sql::SqlConnection* resultDbConn,
+        std::shared_ptr<qmeta::QMetaSelect> const& qMetaSelect,
+        qmeta::CzarId qMetaCzarId,
+        std::string const& userQueryId)
+    : _resultDbConn(resultDbConn),
+      _qMetaSelect(qMetaSelect),
+      _qMetaCzarId(qMetaCzarId),
+      _messageStore(std::make_shared<qdisp::MessageStore>()),
+      _resultTableName(::g_nextResultTableId(userQueryId)) {
+
+    // The SQL statement should be mostly OK alredy but we need to change
+    // table name, instead of INFORMATION_SCHEMA.PROCESSLIST we use special
+    // Qmeta view with the name InfoSchemaProcessList
+    auto stmt = statement->clone();
+    for (auto& tblRef: stmt->getFromList().getTableRefList()) {
+        // assume all table refs have to be replaced
+        // (in practice we accept only one table in FROM
+        tblRef->setDb("");
+        tblRef->setTable("InfoSchemaProcessList");
+    }
+
+    auto qtempl = stmt->getQueryTemplate();
+    _query = qtempl.sqlFragment();
+
+    // we also do not want to return too many results by default
+    // as QMeta can contain a lot of records, limit ourselves
+    // to some reasonable number, users can override with different LIMIT
+    if (! stmt->hasLimit()) {
+        _query += " LIMIT 1000";
+    }
+
+    if (stmt->hasOrderBy()) {
+        _orderBy = stmt->getOrderBy().sqlFragment();
+    }
+}
+
+UserQueryProcessList::UserQueryProcessList(bool full,
+        sql::SqlConnection* resultDbConn,
+        std::shared_ptr<qmeta::QMetaSelect> const& qMetaSelect,
+        qmeta::CzarId qMetaCzarId,
+        std::string const& userQueryId)
+    : _resultDbConn(resultDbConn),
+      _qMetaSelect(qMetaSelect),
+      _qMetaCzarId(qMetaCzarId),
+      _messageStore(std::make_shared<qdisp::MessageStore>()),
+      _resultTableName(::g_nextResultTableId(userQueryId)) {
+
+    // use ShowProcessList view
+    _query = "SELECT Id, User, Host, db, Command, Time, State, ";
+    _query += full ? "Info, " : "SUBSTRING(Info FROM 1 FOR 100) Info, ";
+    // These are non-standard but they need to be there because they appear in WHERE
+    _query += "CzarId, Submitted, Completed";
+    _query += " FROM ShowProcessList";
+
+    // only show stuff for current czar and not too old
+    _query += " WHERE CzarId = " + std::to_string(qMetaCzarId) + " AND "
+            "(Completed IS NULL OR Completed > NOW() - INTERVAL 3 DAY)";
+
```

Comment:
Idea was that `SHOW PROCESSLIST` shows you what you can retrieve from this particular czar. To see stuff from all czars you have to use `SELECT ... FROM INFORMATION_SCHEMA.PROCESSLIST`. We may revisit this when we have better idea how we are going to work with multiple czars.